sdv/model_generator/main.py

```
@@ -30,6 +30,73 @@
+    target_folder: str = "./gen_model",
+    strict: bool = True,
+    overlays: list[str] = [],
+    extended_attributes: str = "",
+):
+    """
+    Generate model using the passed arguments.
+
+    Args:
+        language (str): The target language of the generated code,
+        e.g. [python, cpp],
+        input_file_path (str): The file to convert. Currently supports JSON
+        and Vspec file formats,
+        include_dir (list[str]): Add include directory to search for included
+        vspec files,
+        target_folder str: The folder name (with relative path) where the code
+        will be generated,
+        strict (bool): Use strict checking or not,
+        overlays (list[str]): Add overlays that will be layered on top of
+        the VSS file in the order they,
+        extended-attributes (str): Whitelisted extended attributes
+        as comma separated list
+    """
+    include_dirs = ["."]
+    include_dirs.extend(include_dir)
+
+    ext_attributes_list = extended_attributes.split(",")
+    if len(ext_attributes_list) > 0:
+        vspec.model.vsstree.VSSNode.whitelisted_extended_attributes = (
+            ext_attributes_list
+        )
+        print(f"Known extended attributes: {', '.join(ext_attributes_list)}")
+
+    try:
+        tree = FileImport(input_file_path, include_dirs, strict, overlays).load_tree()
+
+        if language == "python":
+            print("Recursing tree and creating Python code...")
+            VehicleModelPythonGenerator(
+                tree,
+                target_folder,
+                name,
+            ).generate()
+            print("All done.")
+        elif language == "cpp":
+            print("Recursing tree and creating c++ code...")
+            VehicleModelCppGenerator(
+                tree,
+                target_folder,
+                name,
+            ).generate()
+            print("All done.")
+        else:
+            print(f"Language {language} is not supported yet.")
```

The else case should raise an error and exit with a non-zero exit code.